Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.cpp

```diff
@@ -465,14 +468,49 @@ void Wiimote::GetIRData(u8* const data, bool use_accel)
   static const double dist1 = 100.0 / camWidth;  // this seems the optimal distance for zelda
   static const double dist2 = 1.2 * dist1;
 
-  for (auto& vtx : v)
+  float aspect =
+      SConfig::GetInstance().m_SYSCONF->GetData<bool>("IPL.AR") ? 16.0 / 9.0 : 4.0 / 3.0;
+
+  float yScale = bpmem.triggerEFBCopy.scale_invert ? 256.0f / (float)bpmem.dispcopyyscale :
+                                                     (float)bpmem.dispcopyyscale / 256.0f;
+
+  u32 height = static_cast<u32>(1.0f + (float)bpmem.copyTexSrcWH.y * yScale);
+
+  // NOTICE_LOG(WIIMOTE, "aspect ratio : %f", aspect);
+  // NOTICE_LOG(WIIMOTE, "mouse pos /1 : %f-%f", xx, yy);
+
+  if (!g_ActiveConfig.bMiddleSensorBar)
+  {
+    for (auto& vtx : v)
+    {
+      vtx.x = xx * (bndright - bndleft) / 2 + (bndleft + bndright) / 2;
+      if (m_sensor_bar_on_top)
+        vtx.y = yy * (bndup - bnddown) / 2 + (bndup + bnddown) / 2;
+      else
+        vtx.y = yy * (bndup - bnddown) / 2 - (bndup + bnddown) / 2;
+      vtx.z = 0;
+    }
+  }
+  else
   {
-    vtx.x = xx * (bndright - bndleft) / 2 + (bndleft + bndright) / 2;
-    if (m_sensor_bar_on_top)
-      vtx.y = yy * (bndup - bnddown) / 2 + (bndup + bnddown) / 2;
+    if (height != 456)
```

Comment:
Don't hardcode fixes for games based on XFB heights, it's brittle and prone to breaking.